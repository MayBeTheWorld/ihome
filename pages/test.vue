<template>
    <view class="systemBox">
        <status-bar/>
        商城
        <publicTabBar :activePage="2"/>
    </view>
</template>

<script setup>

import StatusBar from '../components/statusBar.vue'
</script>

<style scoped lang="scss">

</style>
