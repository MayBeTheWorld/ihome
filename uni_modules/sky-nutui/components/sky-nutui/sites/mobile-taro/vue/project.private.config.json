{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}