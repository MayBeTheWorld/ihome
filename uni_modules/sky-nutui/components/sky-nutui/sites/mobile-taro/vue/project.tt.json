{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
