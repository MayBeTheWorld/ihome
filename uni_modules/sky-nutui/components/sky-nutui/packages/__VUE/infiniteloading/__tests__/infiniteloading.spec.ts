import { config, mount } from '@vue/test-utils';
import InfiniteLoading from '../index.vue';
import NutIcon from '../../icon/index.vue';
import { trigger, triggerDrag } from '../../../utils/test/event';
import { nextTick, toRefs, reactive, ref, onMounted } from 'vue';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('pull base', async () => {
  const wrapper = mount(InfiniteLoading, {
    props: {
      isOpenRefresh: true,
      loadMoreTxt: '没有更多'
    }
  });
  const track = wrapper.find('.nut-infiniteloading');
  // pulling
  trigger(track, 'touchstart', 0, 0);
  trigger(track, 'touchmove', 0, 20);
  expect(wrapper.html()).toMatchSnapshot();

  // loading
  trigger(track, 'touchend', 0, 100);
  expect(wrapper.html()).toMatchSnapshot();

  // still loading
  triggerDrag(track, 0, 100);

  expect(wrapper.emitted('refresh')).toBeTruthy();
});

test('infiniteloading base', async () => {
  const wrapper = mount({
    components: {
      'nut-infiniteloading': InfiniteLoading,
      'nut-icon': NutIcon
    },
    template: `
    <nut-infiniteloading
      :has-more="hasMore"
      load-more-txt="没有啦～"
      @load-more="loadMore">
        <li class="infiniteLi" v-for="(item, index) in defultList" :key="index">{{item}}</li>
    </nut-infiniteloading>
    `,
    setup() {
      const hasMore = ref(true);
      const data = reactive({
        defultList: []
      });
      const init = () => {
        for (let i = 0; i < 10; i++) {
          (data.defultList as any).push(i);
        }
      };
      const loadMore = (done: any) => {
        setTimeout(() => {
          const curLen = data.defultList.length;
          for (let i = curLen; i < curLen + 10; i++) {
            (data.defultList as any).push(i);
          }
          if (data.defultList.length > 30) hasMore.value = false;
          done();
        }, 500);
      };
      onMounted(() => {
        init();
      });

      return { ...toRefs(data), hasMore, loadMore };
    }
  });
  const track = wrapper.find('.nut-infiniteloading');
  await nextTick();
  trigger(track, 'touchstart', 0, 0);
  trigger(track, 'touchmove', 0, -100);
  trigger(track, 'touchend', 0, -800);

  triggerDrag(track, 0, -800);
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();
});
