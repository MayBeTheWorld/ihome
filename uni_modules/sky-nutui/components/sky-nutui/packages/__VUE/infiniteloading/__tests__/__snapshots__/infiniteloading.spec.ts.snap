// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infiniteloading base 1`] = `
"<view class=\\"nut-infiniteloading\\">
  <view class=\\"nut-infinite-top\\" style=\\"height: 0px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);\\">
    <view class=\\"top-box\\"><img class=\\"nut-icon__img top-img\\" src=\\"https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png\\">
      <view class=\\"top-text\\">松开刷新</view>
    </view>
  </view>
  <view class=\\"nut-infinite-container\\">
    <li class=\\"infiniteLi\\">0</li>
    <li class=\\"infiniteLi\\">1</li>
    <li class=\\"infiniteLi\\">2</li>
    <li class=\\"infiniteLi\\">3</li>
    <li class=\\"infiniteLi\\">4</li>
    <li class=\\"infiniteLi\\">5</li>
    <li class=\\"infiniteLi\\">6</li>
    <li class=\\"infiniteLi\\">7</li>
    <li class=\\"infiniteLi\\">8</li>
    <li class=\\"infiniteLi\\">9</li>
  </view>
  <view class=\\"nut-infinite-bottom\\">
    <!--v-if-->
  </view>
</view>"
`;

exports[`pull base 1`] = `
"<view class=\\"nut-infiniteloading\\">
  <view class=\\"nut-infinite-top\\" style=\\"height: 0px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);\\">
    <view class=\\"top-box\\"><img class=\\"nut-icon__img top-img\\" src=\\"https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png\\">
      <view class=\\"top-text\\">松开刷新</view>
    </view>
  </view>
  <view class=\\"nut-infinite-container\\"></view>
  <view class=\\"nut-infinite-bottom\\">
    <!--v-if-->
  </view>
</view>"
`;

exports[`pull base 2`] = `
"<view class=\\"nut-infiniteloading\\">
  <view class=\\"nut-infinite-top\\" style=\\"height: 0px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);\\">
    <view class=\\"top-box\\"><img class=\\"nut-icon__img top-img\\" src=\\"https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png\\">
      <view class=\\"top-text\\">松开刷新</view>
    </view>
  </view>
  <view class=\\"nut-infinite-container\\"></view>
  <view class=\\"nut-infinite-bottom\\">
    <!--v-if-->
  </view>
</view>"
`;
